modules/reactor/include/meshgenerators/ReactorGeometryMeshBuilderBase.h

```diff
@@ -41,6 +84,63 @@ class ReactorGeometryMeshBuilderBase : public MeshGenerator
    */
   void initializeReactorMeshParams(const std::string reactor_param_name);
 
+  /**
+   * Print metadata associated with ReactorGeometryMeshBuilder object
+   * @param geometry_type       type of geometry (pin / assembly / core) under consideration
+   * @param mg_name             name of mesh generator associated with this object
+   * @param first_function_call whether this is the original function call, which will trigger
+   * additional output messages
+   */
+  void printReactorMetadata(const std::string geometry_type,
+                            const std::string mg_name,
+                            const bool first_function_call = true);
+
+  /**
+   * Print core-level metadata associated with ReactorGeometryMeshBuilder object
+   * @param mg_name name of mesh generator associated with core
+   * @param first_function_call whether this is the original function call, which will trigger
+   * additional output messages
+   */
+  void printCoreMetadata(const std::string mg_name, const bool first_function_call);
+
+  /**
+   * Print assembly-level metadata associated with ReactorGeometryMeshBuilder object
+   * @param mg_name name of mesh generator associated with assembly
+   * @param whether this is the original function call, which will trigger additional output
+   * messages
+   */
+  void printAssemblyMetadata(const std::string mg_name, const bool first_function_call);
+
+  /**
+   * Print pin-level metadata associated with ReactorGeometryMeshBuilder object
+   * @param mg_name name of mesh generator associated with assembly
+   */
+  void printPinMetadata(const std::string mg_name);
+
+  /**
+   * Print global ReactorMeshParams metadata associated with ReactorGeometryMeshBuilder object
+   */
+  void printGlobalReactorMetadata();
```

Comment:
big thumbs up for all this work printing the data and making it possible for users to self-help by looking at what is going on behind the scenes, within reason

Reply:
Thanks!